src/ape/api/projects.py

```diff
@@ -123,17 +123,36 @@ def _create_source_dict(
         cls, contract_paths: Collection[Path], base_path: Path
     ) -> Dict[str, Source]:
         return {
-            str(get_relative_path(source, base_path)): Source(  # type: ignore
+            str(get_relative_path(source_path, base_path)): Source(  # type: ignore
                 checksum=Checksum(  # type: ignore
                     algorithm="md5",
-                    hash=compute_checksum(source.read_bytes()),
+                    hash=compute_checksum(source_path.read_bytes()),
                 ),
                 urls=[],
-                content=source.read_text(),
+                content=source_path.read_text(),
+                references=cls._create_source_reference_list(source_path=source_path),
             )
-            for source in contract_paths
+            for source_path in contract_paths
         }
 
+    @classmethod
+    def _create_source_reference_list(cls, source_path: Path) -> List[str]:
```

Comment:
Was picturing that this would be an API method, where all the paths are given as an input, and a Dict is returned with the references of each path to the set of references that depend on them. That would get stored in the manifest, and then referenced whenever changes are detected in the project manager to expand the set of files that need to be recompiled.

So basically, this logic should live under the solidity plugin.

Because this would be a breaking change, maybe the default would be to return a Dict with empty sets of references